lib/NNlibCUDA/test/upsample.jl

```diff
@@ -0,0 +1,26 @@
+@testset "Bilinear upsampling" begin
+    x = Float32[1 2; 3 4][:,:,:,:]
+    x = cat(x,x; dims=3)
+    x = cat(x,x; dims=4)
+    x = cu(x)
+  
+    y_true = Float32[ 1//1  4//3   5//3   2//1;
+            7//5 26//15 31//15 12//5;
+            9//5 32//15 37//15 14//5;
+           11//5 38//15 43//15 16//5;
+           13//5 44//15 49//15 18//5;
+            3//1 10//3  11//3   4//1]
+    y_true = cat(y_true,y_true; dims=3)
+    y_true = cat(y_true,y_true; dims=4)
+    y_true = cu(y_true)
+  
+    @allowscalar y = upsample_bilinear(x, (3,2))
```

Comment:
@maxfreu this would error out when `allowscalar(false)`, is that normal?

Reply:
Please mark that as broken for rhe time being.

Reply:
I'm not sure this is broken, maybe it is just the way it is and scalar indexing is not impacting performance in this case. @maxfreu will know more since he implemented it

Reply:
Since our requirement is for the suite to not perform scalar indexing, and it errors in that state, I think it's fair to say something needs correcting. We can investigate the cause for that later, and good to intimate the author.

Reply:
You have encountered a problem this very PR is going to fix: You have to be on CUDA#master, since there has not been a release including the underlying functions yet. So here it falls back to the CPU implementation, since it finds nothing better.

Reply:
my bad, I wasn't using the kernel I just ported, fixed now